aptos-move/framework/aptos-framework/sources/object.move

```diff
@@ -725,6 +726,17 @@ module aptos_framework::object {
         )
     }
 
+    /// Grant a transfer permission to the permissioned signer using TransferRef.
+    public fun grant_permission_with_transfer_ref(
+        permissioned_signer: &signer,
+        ref: &TransferRef,
+    ) {
```

Comment:
we should consider having also the function with_constructor_ref, for convenience, even though you can always get transfer_ref from it

though have no strong feelings about it